src/wp-admin/upload.php

```diff
@@ -141,7 +141,13 @@
 	wp_enqueue_script( 'media-grid' );
 	wp_enqueue_script( 'media' );
 
-	remove_action( 'admin_head', 'wp_admin_canonical_url' );
+	// Remove the error parameter added by deprecation of wp-admin/media.php.
+	add_filter(
+		'removable_query_args',
+		function( $query_args ) {
+				return array( 'error' );
+		}
+	);
```

Comment:
```suggestion
	add_filter(
		'removable_query_args',
		function() {
			return array( 'error' );
		},
		10,
		0
	);
```
This lets us remove the parameter and ensure none are ever passed to it.